mackerel-plugin.go

```diff
@@ -124,7 +124,12 @@ func (h *MackerelPlugin) saveValues(values map[string]interface{}, now time.Time
 	if !h.hasDiff() {
 		return nil
 	}
-	f, err := os.Create(h.tempfilename())
+	fname := h.tempfilename()
+	err := os.MkdirAll(filepath.Dir(fname), 0777)
```

Comment:
We may skip `MkdirAll` and simply return "directory not found" error?
(To me, `MkdirAll` is too aggressive, because with this implementation when someone specifies wrong `MACKEREL_PLUGIN_WORKDIR`, the wrong directory will be created)